tests/integration/components/repository-status-toggle-test.js

```diff
@@ -26,4 +27,66 @@ module('RepositoryStatusToggleComponent', function (hooks) {
 
     assert.dom('.switch').hasClass('active', 'switch should have active class');
   });
+
+  test('should display correct error message on 409 fail', async function (assert) {
+    let store = this.owner.lookup('service:store');
+    let repo = store.push({
+      data: {
+        id: 10001,
+        type: 'repo',
+        attributes: {
+          slug: 'travis-ci/travis-web',
+          active: false,
+          permissions: {
+            admin: true
+          },
+        }
+      }
+    });
+
+    this.set('repository', repo);
+
+    server.post('/repo/:id/activate', (schema, request) => {
+      return new Response(409, {}, {});
+    });
+
+    await render(hbs`{{repository-status-toggle repository=repository}}`);
+    assert.dom('.switch').findElement().click();
+    settled().then(() => {
+      assert.dom('.repositories-error').hasText(
+        'Request cannot be completed because the repository ssh key is still pending to be created. Please retry in a bit, or try syncing the repository if this condition does not resolve.'
+      );
+    });
+  });
+
+  test('should display correct error message on non 409 fail', async function (assert) {
+    let store = this.owner.lookup('service:store');
+    let repo = store.push({
+      data: {
+        id: 10001,
+        type: 'repo',
+        attributes: {
+          slug: 'travis-ci/travis-web',
+          active: false,
+          permissions: {
+            admin: true
+          },
+        }
+      }
+    });
+
+    this.set('repository', repo);
+
+    server.post('/repo/:id/activate', (schema, request) => {
+      return new Response(404, {}, {});
+    });
+
+    await render(hbs`{{repository-status-toggle repository=repository}}`);
+    assert.dom('.switch').findElement().click();
+    settled().then(() => {
+      assert.dom('.repositories-error').hasText(
+        'An error happened when we tried to alter settings on GitHub. It may be caused by API restrictions, please review and add your authorized Orgs.'
+      );
+    });
+  });
```

Comment:
I meant not integration but acceptance test, like this one: https://github.com/travis-ci/travis-web/blob/master/tests/acceptance/profile/update-repositories-test.js

They are the most important ones, as soon as they run the entire application and execute real user scenarios, whereas integration tests (like this one) run just a shallow component instance in isolation.